business-networks/freighter-tests/src/freighterTest/kotlin/freightertests/BaseBNFreighterTest.kt

```diff
@@ -0,0 +1,204 @@
+    private val freighterHelperCordapp = NodeBuilder.DeployedCordapp.fromClassPath("freighter-cordapp-flows")
+
+    abstract fun getLogger(): ExtendedLogger
+    abstract fun runScenario(numberOfParticipants: Int, deploymentContext: DeploymentContext): Map<String, Long>
+
+
+    fun runBenchmark(numberOfParticipants: Int,cutOffTime:Long){
+        val deploymentContext = DeploymentContext(machineProvider, nms, artifactoryUsername, artifactoryPassword)
+        val benchmark = runScenario( numberOfParticipants,deploymentContext)
+
+        benchmark.map {
+            getLogger().info("${it.key} BenchMark ${it.value}")
+        }
+
+        benchmark.map {
+            Assertions.assertTrue(it.value <= cutOffTime, "${it.value} exceeded the benchmark of $cutOffTime")
+        }
+    }
+
+    fun checkGroupSizeIsAsExpectedInMembersVaults(listOfNetworkMembers: List<SingleNodeDeployed>, groupLinearId: UniqueIdentifier, expectedGroupSize: Int) {
+        val listToIterateOver = listOfNetworkMembers.toMutableList()
+        while (listToIterateOver.isNotEmpty()) {
+            val itr = listToIterateOver.iterator()
+            while (itr.hasNext()) {
+                val currentNode = itr.next()
+                val actual = currentNode.rpc {
+                    vaultQuery(GroupState::class.java).states.map { it.state.data }
+                }
+
+                for (groupState in actual) {
+                    if (groupState.linearId == groupLinearId && groupState.participants.size == expectedGroupSize) {
+                        getLogger().info("${currentNode.name} has correct group size")
+                        itr.remove()
+                    } else if (groupState.linearId == groupLinearId) {
+                        getLogger().info(currentNode.name + "still waiting on full group up")
+                        getLogger().info("Group members: ${groupState.participants.size} out of $expectedGroupSize")
+                    }
+                }
+                if (actual.isEmpty() || actual.first().linearId != groupLinearId) {
+                    getLogger().error("${currentNode.name} does not have the expected group state in vault")
+                }
+            }
+        }
+    }
+
+    fun addMembersToAGroup(bnoNode: SingleNodeDeployed, newGroupId: UniqueIdentifier, newGroupName: String, groupMembers: MutableList<UniqueIdentifier>) {
+        bnoNode.rpc {
+            startFlow(::ModifyGroupFlow,
+                    newGroupId,
+                    newGroupName,
+                    groupMembers.toMutableSet(),
+                    null).returnValue.getOrThrow()
+        }
+    }
+
+    fun activateNetworkMembership(nodeToMembershipIds: Map<SingleNodeDeployed, MembershipState>, bnoNode: SingleNodeDeployed) {
+        nodeToMembershipIds.map {
+            bnoNode.rpc {
+                startFlow(::ActivateMembershipFlow,
+                        it.value.linearId,
+                        null)
+            }
+        }
```

Comment:
Should't you wait for all activate membership flows to get return value:
```suggestion
        nodeToMembershipIds.map {
            bnoNode.rpc {
                startFlow(::ActivateMembershipFlow,
                        it.value.linearId,
                        null)
            }
        }.map {
            it.getOrThrow()
        }
```